components/influxdb/scripts/start.py

```diff
@@ -17,7 +19,47 @@
 
 
 INFLUXDB_ENDPOINT_IP = ctx_properties['influxdb_endpoint_ip']
+INFLUXDB_ENDPOINT_PORT = 8086
+RETRIES = 10
+
+
+def is_influxdb_responding(url):
+    """Check if url responds with a normal influxdb response."""
+    try:
+        urllib2.urlopen(url).read()
+    except urllib2.HTTPError as e:
+        # influxdb normally responds with a 404 on GET to /
+        if e.code != 404:
+            return False
+        return True
+    except urllib2.URLError:
+        return False
+    else:
+        return True
+
 
 if not INFLUXDB_ENDPOINT_IP:
     ctx.logger.info('Starting InfluxDB Service...')
     utils.start_service_and_archive_properties(INFLUX_SERVICE_NAME)
+
+    INFLUXDB_ENDPOINT_IP = '127.0.0.1'
```

Comment:
this entire section below looks like a duplication of the elasticsearch one. Why not create a utils class which is instantiated for the relevant service and performs the relevant checks?